src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -3639,6 +3649,810 @@ .maxstack  3
+        public void CopyCtor_WithInitializers()
+        {
+            var source =
+@"public record C(object N1, object N2)
+{
+    private int field = 42;
+    public int Property = 43;
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+
+            var verifier = CompileAndVerify(comp, verify: ExecutionConditionUtil.IsCoreClr ? Verification.Skipped : Verification.Fails);
+            verifier.VerifyIL("C..ctor(C)", @"
+{
+  // Code size       55 (0x37)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  call       ""object..ctor()""
+  IL_0006:  ldarg.0
+  IL_0007:  ldarg.1
+  IL_0008:  ldfld      ""object C.<N1>k__BackingField""
+  IL_000d:  stfld      ""object C.<N1>k__BackingField""
+  IL_0012:  ldarg.0
+  IL_0013:  ldarg.1
+  IL_0014:  ldfld      ""object C.<N2>k__BackingField""
+  IL_0019:  stfld      ""object C.<N2>k__BackingField""
+  IL_001e:  ldarg.0
+  IL_001f:  ldarg.1
+  IL_0020:  ldfld      ""int C.field""
+  IL_0025:  stfld      ""int C.field""
+  IL_002a:  ldarg.0
+  IL_002b:  ldarg.1
+  IL_002c:  ldfld      ""int C.Property""
+  IL_0031:  stfld      ""int C.Property""
+  IL_0036:  ret
+}");
+        }
+
+        [Fact, WorkItem(44902, "https://github.com/dotnet/roslyn/issues/44902")]
+        public void CopyCtor_UserDefinedButDoesNotDelegateToBaseCopyCtor()
+        {
+            var source =
+@"public record B(object N1, object N2)
+{
+}
+public record C(object P1, object P2) : B(0, 1)
+{
+    public C(C c) // 1, 2
+    {
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (6,12): error CS1729: 'B' does not contain a constructor that takes 0 arguments
+                //     public C(C c) // 1, 2
+                Diagnostic(ErrorCode.ERR_BadCtorArgCount, "C").WithArguments("B", "0").WithLocation(6, 12),
+                // (6,12): error CS8868: A copy constructor in a record type deriving from another record must invoke that base record's copy constructor 'B.B(B)'.
+                //     public C(C c) // 1, 2
+                Diagnostic(ErrorCode.ERR_CopyConstructorMustInvokeBaseCopyConstructor, "C").WithArguments("B.B(B)").WithLocation(6, 12)
+                );
+        }
+
+        [Fact, WorkItem(44902, "https://github.com/dotnet/roslyn/issues/44902")]
+        public void CopyCtor_UserDefinedButDoesNotDelegateToBaseCopyCtor_DerivesFromObject()
+        {
+            var source =
+@"public record C(int I)
+{
+    public C(C c)
+    {
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+            var verifier = CompileAndVerify(comp, verify: ExecutionConditionUtil.IsCoreClr ? Verification.Skipped : Verification.Fails);
+            verifier.VerifyIL("C..ctor(C)", @"
+{
+  // Code size       14 (0xe)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  ldarg.1
+  IL_0002:  stfld      ""int C.<I>k__BackingField""
```

Comment:
This looks wrong. We are trying to store value of type ```C``` into a field of type ```int```. Why this test didn't fail? Success scenarios should be executed to ensure complete verification. BTW, it looks like we are including initializers into the copy constructor, which contradicts what this PR description is saying: "don't run initializers in synthesized copy constructor"

Reply:
This is not a synthesized copy constructor, thus the field initializer is included. We're putting `0` into the backing field, which has type `int`. None of this was affected by the current PR, which only potentially adds a diagnostic on such user-defined copy constructors.

---
In reply to: [439020094](https://github.com/dotnet/roslyn/pull/45022#discussion_r439020094) [](ancestors = 439020094)

Reply:
> This is not a synthesized copy constructor, thus the field initializer is included. We're putting 0 into the backing field, which has type int. None of this was affected by the current PR, which only potentially adds a diagnostic on such user-defined copy constructors.

This is a violation of the language design. A copy constructor is not supposed to run initializers

---
In reply to: [439058957](https://github.com/dotnet/roslyn/pull/45022#discussion_r439058957) [](ancestors = 439058957,439020094)

Reply:
Okay. The notes I sent following the meeting were incorrect. Sent correction

---
In reply to: [439066375](https://github.com/dotnet/roslyn/pull/45022#discussion_r439066375) [](ancestors = 439066375,439058957,439020094)

Reply:
Consider adding similar test without primary ctor, but with an explicit field initializer.

---
In reply to: [439072053](https://github.com/dotnet/roslyn/pull/45022#discussion_r439072053) [](ancestors = 439072053,439066375,439058957,439020094)

Reply:
> Consider adding similar test without primary ctor, but with an explicit field initializer.

This was regarding CopyCtor_UserDefinedButDoesNotDelegateToBaseCopyCtor_DerivesFromObject test

---
In reply to: [439103373](https://github.com/dotnet/roslyn/pull/45022#discussion_r439103373) [](ancestors = 439103373,439072053,439066375,439058957,439020094)